c++/src/capnp/serialize-test.c++

```diff
@@ -476,6 +476,15 @@ TEST(Serialize, RejectHugeMessage) {
 }
 #endif  // !__MINGW32__
 
+TEST(Serialize, SegmentsTable) {
+  TestMessageBuilder builder(1);
+  initTestMessage(builder.initRoot<TestAllTypes>());
+
+  kj::Array<byte> table = segmentsTable(builder.getSegmentsForOutput());
+  byte expected[] = { 0, 0, 0, 0, 95, 1, 0, 0 };
```

Comment:
These bytes could change if `TestAllTypes` grows new fields.

Perhaps a more robust approach would be to serialize the message the normal way and verify that it leads to the same bytes as serializing the segment table and concatenating with the segments.